### DISCUSSION TITLE
windows server 2019 do not respond for any nvm commands during remote session or when execute over azure devops pipeline task

### DISCUSSION THREAD

**Discussion by CNManoj** [coreybutler/nvm-windows#1112] [General]
**Operating system and version:** Windows Server 2019 Std
**nvm debug output:**
Details
Windows PowerShell Windows Version: 10.0 (Build 17763)
Windows Developer Mode: UNKNOWN (user cannot read registry)

NVM4W Version: 1.1.12
NVM4W Path: C:\ProgramData\nvm\nvm.exe
NVM4W Settings: C:\ProgramData\nvm\settings.txt
NVM_HOME: C:\ProgramData\nvm
NVM_SYMLINK: C:\Program Files\nodejs
Node Installations: C:\ProgramData\nvm

Total Node.js Versions: 1
Active Node.js Version: v20.11.1

No problems detected.

Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues

**nvm ls output:**
Details
* 21.6.2 (Currently using 64-bit executable)

**How did you install nvm?** 
using nvm-windows (nvm-setup.exe) version 1.1.12

**What steps did you perform?** 
Double click installed from nvm-setp.exe, then **_npm install latest_**

**What is working?** 
Installed and configured nvm and can use locally from server using powershell and cmd

**What is not working?** 
nvm commands to work from azure devops pipeline or PSSession remote but it doesn't respond for ever

**Azure DevOps Version and Agent used:**
Azure DevOps Server 2019.1.1
![Log-PS-NVM_Timeout](https://github.com/coreybutler/nvm-windows/assets/88138390/e213cf1e-2ac6-4cab-8d85-8821ce8e440b)
![PS-NVM_Timeout](https://github.com/coreybutler/nvm-windows/assets/88138390/4c828212-2225-4d90-a525-3b1e46848dc9)

Agent version used: 2.200.0 
NVM version: 1.1.12

**Everything works with:**
Azure DevOps Server 2019.1.1
Agent version used: 2.200.0
NVM version: 1.1.11

**Comment by coreybutler** [comment]
Unfortunately, I do not have an environment like this to test in and there isn't enough information to troubleshoot with what's provided. However; I suspect it has to do with https://github.com/coreybutler/nvm-windows/issues/1068

**Comment by Mhbuur** [comment]
This is indeed related to #1068 
An azure pipeline command line task  with only the script "nvm use 22" hangs forever, and there are no error messages. The command will execute just fine if you go to the agent server as the service user and execute it.

The workaround is to launch a new powershell console, like so:
`start powershell "nvm use 22"`

#1120 seems to fix it, but has not been approved.
